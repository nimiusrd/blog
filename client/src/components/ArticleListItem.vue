<template>
    <li class="item" @click="handleClick">
        <div class="title">
            {{ title }}
        </div>
        <div class="author">
            {{ author }}
        </div>
        <div class="posted-at">
            {{ formattedPostedAt }}
        </div>
        <div class="updated-at">
            {{ formattedUpdatedAt }}
        </div>
    </li>
</template>

<script lang="ts">
import Vue from 'vue'

const ArticleListItem = Vue.extend({
    props: {
        id: Number,
        title: String,
        author: String,
        postedAt: String,
        updatedAt: String
    },
    computed: {
        formattedPostedAt: function() {
            return new Date(this.postedAt).toDateString()
        },
        formattedUpdatedAt: function() {
            return new Date(this.updatedAt).toDateString()
        },
    },
    methods: {
        handleClick() {
            this.$router.push({
                path: `/articles/${this.id}`
            })
        }
    }
})

export default ArticleListItem
</script>

<style scoped>
.item {
    display: grid;
    font-size: 16px;
    font-weight: 400;
    grid-template:
        "title author posted-at updated-at"
        / 3fr 1fr 1fr 1fr;
    text-align: center;
}

.item:hover {
    background: rgba(241, 243, 244);
    cursor: pointer;
}

.item:hover:active {
    background: rgba(231, 233, 235);
    cursor: pointer;
}

.title {
    border: 1px black solid;
    grid-area: title;
    padding: 16px;
}

.author {
    border: 1px black solid;
    grid-area: author;
    padding: 16px;
}

.posted-at {
    border: 1px black solid;
    grid-area: posted-at;
    padding: 16px;
}

.updated-at {
    border: 1px black solid;
    grid-area: updated-at;
    padding: 16px;
}
</style>
