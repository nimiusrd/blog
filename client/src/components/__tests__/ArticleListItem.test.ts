import ArticleListItem from '@/components/ArticleListItem.vue'
import {shallowMount} from '@vue/test-utils'

test('snapshot', () => {
    const wrapper = shallowMount<ArticleListItem>(ArticleListItem, {
        propsData: {
            id: 1,
            title: 'title',
            author: 'author',
            postedAt: '2019-09-28T06:34:56+00:00',
            updatedAt: '2019-09-29T05:03:43+00:00'
        }
    })

    expect(wrapper.html()).toMatchSnapshot()
})
