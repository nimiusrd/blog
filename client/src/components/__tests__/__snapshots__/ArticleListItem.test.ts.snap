// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
"<li class=\\"item\\"><div class=\\"title\\">
        title
    </div> <div class=\\"author\\">
        author
    </div> <div class=\\"posted-at\\">
        Sat Sep 28 2019
    </div> <div class=\\"updated-at\\">
        Sun Sep 29 2019
    </div></li>"
`;
